File: backend/myapp/core/dlog.py
# pyright: strict
import logging
import traceback
# setup logger
__logger = logging.getLogger(__name__)
__logger.setLevel("DEBUG")
handler = logging.StreamHandler()
log_format = "[%(asctime)s][%(levelname)s] %(message)s"
formatter = logging.Formatter(log_format)
handler.setFormatter(formatter)
__logger.addHandler(handler)


def stack():
    print("*" * 100)
    traceback.print_stack(limit=5)
    print("*" * 100)


def d(msg: str):
    __logger.debug(msg)


def e(msg: str):
    __logger.error(msg)


def ex(e: Exception, msg: str = "Exception:"):
    __logger.error("{} - {}".format(msg, e.args))
    __logger.exception("{} - {}".format(msg, e.args), e)

File: packages/www2/src/libs/dlog.ts
export namespace dlog {
  export function d(msg?: string) {
    console.log(`\n[DEBUG]${msg || "null"}`);
  }
  export function e(msg?: string) {
    console.log(`[ERROR]${msg || "null"}`);
  }
  export function obj(obj?: Object, msg: string = "DEBUG OBJECT") {
    console.log(`=========================== ${msg} START ======================`);
    console.log(JSON.stringify(obj || {}, null, 1));
    console.log(`=========================== ${msg} END ======================`);
  }
  export function map(map: Map<string, any>) {
    console.log("=========================== DEBUG OBJECT  START ======================");
    for (let x in map.keys()) {
      console.log(`${JSON.stringify(x)} ==> ${JSON.stringify(map.get(x))}`);
    }
    console.log("=========================== DEBUG OBJECT END ======================");
  }
  export function ex(e: Error) {
    console.log(`[ERROR]${e.message}`);
    console.log(e.stack);
  }
}
